docroot/sites/all/modules/dev/warmshowers_site/warmshowers_site.module

```diff
@@ -74,57 +161,62 @@ function warmshowers_site_block_info() {
  * @return array
  */
 function warmshowers_site_block_view($delta) {
+  global $user;
   $block = array();
 
   switch ($delta) {
     case 'ws_user_account_menu':
-      $block['subject'] = '';
+      $block['subject'] = '<none>';
 
-      $block['content'] = _warmshowers_site_generate_ws_user_account_menu();
+      $block['content'] = !empty($user->uid) ? theme('ws_user_account_menu') : '';
+      break;
+    case 'ws_anonymous_user_menu':
+      $block['subject'] = '<none>';
+
+      $block['content'] = empty($user->uid) ? theme('ws_anonymous_user_menu') : '';
+      break;
+    case 'ws_anonymous_welcome_message':
+      $block['subject'] = '<none>';
+
+      $block['content'] = theme('ws_anonymous_welcome_block');
+      break;
+    case 'ws_member_stats_block':
+      $block['subject'] = '<none>';
+
+      $block['content'] = theme('ws_member_stats_block');
       break;
   }
   return $block;
 }
 
 /**
- * Helper function to generate the content for the user account menu block.
+ * Implements hook_preprocess().
  *
- * @return string
+ * Provides data to theme functions.
+ *
+ * @param $variables
+ * @return mixed
  */
-function _warmshowers_site_generate_ws_user_account_menu() {
-  global $user;
-  $content = '';
-
-  if (!empty($user->uid)) {
-    $content = t("Logged in as !name | !logout", array(
-      '!name' => l($user->data['fullname'], 'user/' . $user->uid),
-      '!logout' => l(t('Log out'), 'user/logout')
-    ));
-  }
-  else {
-    $content = l(t('Sign up'), 'user/register', array(
-      'attributes' => array(
-        'class' => array('signup')
-      )
-    ));
-    $content .= l(t('Log in'), 'user', array(
-      'attributes' => array(
-        'class' => array('login')
-      )
-    ));
+function warmshowers_site_preprocess(&$variables, $key) {
+
+  switch($key) {
+    case 'ws_anonymous_user_menu':
```

Comment:
I don't really like calling something a "menu" when it's not a drupal menu. 

Is this breaking existing translations? It can if necessary, just wondering.